tests/phpunit/Unit/functions/StrToCamelCaseTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Cig\Tests\Unit;
+
+class StrToCamelCaseTest extends \Cig\Tests\Unit\BaseTestCase {
+
+	public function test_str_to_camel_case(): void {
+		$string = "welcome~ to camel' case!!";
+		$expected_result = "welcomeToCamelCase";
+
+		//this method uses another method (str_to_words) from same file (string.php)
+		$result = \Cig\str_to_camel_case($string);
+
+		self::assertIsString($result);
+		self::assertSame($expected_result, $result);
+		// skipping international alphabet tests
+	}
+
+	//TODO: expected this to break but this method goes ahead and turns
+	// numbers into strings despite typescript asking for a string
+	public function test_number_to_camel_case(): void {
+		$not_a_string = 101;
```

Comment:
If it's throwing an error, that's likely due to running an older version of PHP. The ability to type class properties was added in 7.4 so I'd make sure you're running on that.
As for typing variables, I'm not certain if PHP 8.0 supports that, but I wouldn't worry about it for now. You can use the class properties and then reference them as needed in the methods. 
If that doesn't make sense we can jump on zoom and walk through it. Might be easier to see it than explain.

As far as why it's not throwing an exception I'm not 100% sure. I'm guessing PHP is trying to be helpful and converting the type. If that's the case using a class property with a specific type may cause the exception to be thrown. Otherwise or in addition to that you can add `declare(strict_types=1);` at the top of the file to see if that prevents any type conversion from happening.
I know options aren't always the best, but here's one more. If you don't like the class property idea you can use `settype($not_a_string, 'int')` on the line below the variable declaration to make sure the type is specified.